dpytools/s3/basic.py

```diff
@@ -45,31 +43,23 @@ def read_s3_file_content_as_dict(
     """
     if not object_name.endswith(".json"):
         raise ValueError("Object name must end with '.json'")
```

Comment:
Should we unit test this error? It doesn't seem significant but I don't know how much coverage we want on this.

Reply:
to the wider point the ideal is 100% coverage ...but.... there is a fine line between sensible coverage and wasting time on edge cases.

I trend towards cover the main branches of logic (so expected usage plus exceptions) with an eye on filling out coverage when we get to the cleaning up phase of the current tranche of work.